tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -4649,22 +4735,25 @@ SpirvInstruction *SpirvEmitter::doUnaryOperator(const UnaryOperator *expr) {
 
     const spv::Op spvOp = translateOp(isInc ? BO_Add : BO_Sub, subType);
     SpirvInstruction *originValue =
-        subValue->isRValue() ? subValue
-                             : spvBuilder.createLoad(subType, subValue);
+        subValue->isRValue()
+            ? subValue
+            : spvBuilder.createLoad(subType, subValue, expr->getLocStart());
     auto *one = hlsl::IsHLSLMatType(subType) ? getMatElemValueOne(subType)
                                              : getValueOne(subType);
 
     SpirvInstruction *incValue = nullptr;
     if (isMxNMatrix(subType)) {
       // For matrices, we can only increment/decrement each vector of it.
-      const auto actOnEachVec = [this, spvOp, one](uint32_t /*index*/,
-                                                   QualType vecType,
-                                                   SpirvInstruction *lhsVec) {
-        auto *val = spvBuilder.createBinaryOp(spvOp, vecType, lhsVec, one);
+      const auto actOnEachVec = [this, spvOp, one,
+                                 expr](uint32_t /*index*/, QualType vecType,
+                                       SpirvInstruction *lhsVec) {
+        auto *val = spvBuilder.createBinaryOp(spvOp, vecType, lhsVec, one,
+                                              expr->getOperatorLoc());
         val->setRValue();
         return val;
       };
-      incValue = processEachVectorInMatrix(subExpr, originValue, actOnEachVec);
+      incValue = processEachVectorInMatrix(subExpr, originValue, actOnEachVec,
+                                           expr->getLocStart());
```

Comment:
should this be `subExpr->getLocStart()`? if so, the callee (`processEachVectorInMatrix`) can find this information from `subExpr`, and there will be no need for the additional parameter passed to the function.